src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryArgumentSetterConfig.java

```diff
@@ -0,0 +1,269 @@
+  }
+
+  private void validateArgumentsSelectionConditions(
+      String argumentSelectionConditions, FailureCollector collector) {
+    if (containsMacro(NAME_ARGUMENT_SELECTION_CONDITIONS)) {
+      return;
+    }
+
+    if (Strings.isNullOrEmpty(argumentSelectionConditions)) {
+      collector.addFailure(
+          "Arguments Selection Conditions is empty!",
+          "Arguments Selection condition can not be empty");
+      return;
+    }
+
+    try {
+      getArgumentSelectionConditionsMap();
+    } catch (Exception e) {
+      collector.addFailure(
+          e.getMessage(), "Invalid key value pair for Argument Selection Conditions");
+    }
+  }
+
+  private void validateArgumentsColumns(String argumentsColumns, FailureCollector collector) {
+    if (containsMacro(NAME_ARGUMENTS_COLUMNS)) {
+      return;
+    }
+    if (Strings.isNullOrEmpty(argumentsColumns)) {
+      collector.addFailure("Arguments Columns is empty!", "Arguments Columns can not be empty");
+    }
+  }
+
+  public QueryJobConfiguration getQueryJobConfiguration() {
+    Table sourceTable = BigQueryUtil.getBigQueryTable(project, dataset, table, serviceFilePath);
+
+    StandardTableDefinition tableDefinition = Objects.requireNonNull(sourceTable).getDefinition();
+    FieldList fields = Objects.requireNonNull(tableDefinition.getSchema()).getFields();
+
+    Map<String, String> argumentConditionMap = getArgumentSelectionConditionsMap();
+    Map<String, Field> argumentConditionFields =
+        extractArgumentsFields(fields, argumentConditionMap);
+
+    checkIfArgumentsColumnsExitsInSource(argumentConditionMap, argumentConditionFields);
+
+    String selectClause = getSelectClause();
+    String whereCondition = getWhereCondition(argumentConditionMap.keySet());
+    String tableName = dataset + "." + table;
+    String query = String.format(QUERY_TEMPLATE, selectClause, tableName, whereCondition);
+
+    Builder queryJobConfiguration = QueryJobConfiguration.newBuilder(query);
+    getParametersValues(argumentConditionMap.entrySet(), argumentConditionFields)
+        .forEach(
+            stringQueryParameterValueSimpleEntry ->
+                queryJobConfiguration.addNamedParameter(
+                    stringQueryParameterValueSimpleEntry.getKey(),
+                    stringQueryParameterValueSimpleEntry.getValue()));
+
+    return queryJobConfiguration.build();
+  }
+
+  private void checkIfArgumentsColumnsExitsInSource(
+      Map<String, String> argumentConditionMap, Map<String, Field> argumentConditionFields) {
+    if (argumentConditionMap.size() == argumentConditionFields.size()) {
+      return;
+    }
+    String nonExistingColumnNames =
+        argumentConditionMap.keySet().stream()
+            .filter(columnName -> !argumentConditionFields.containsKey(columnName))
+            .collect(Collectors.joining(" ,"));
+    throw new RuntimeException(
+        String.format("Those Columns Name \" %s \"do not exits in Table", nonExistingColumnNames));
```

Comment:
`Columns: %s do not exist in table.`
What is the corrective action that user needs to take in this case?